luaui/barwidgets.lua

```diff
@@ -899,15 +920,25 @@ end
 --------------------------------------------------------------------------------
 
 function widgetHandler:IsWidgetKnown(name)
-	return self.knownWidgets[name] and true or false
+	return self.knownWidgetInfos[name] and true or false
 end
 
 function widgetHandler:EnableWidget(name, enableLocalsAccess)
-	local ki = self.knownWidgets[name]
+	local ki = self.knownWidgetInfos[name]
 	if not ki then
 		Spring.Echo("EnableWidget(), could not find widget: " .. tostring(name))
 		return false
 	end
+
+	-- make sure parent is enabled first
+	if ki.parent and not ki.parent.active then
+		self:EnableWidget(ki.parent.name, enableLocalsAccess)
+		--if not self:EnableWidget(ki.parent.name, enableLocalsAccess) then
+		--	Spring.Echo('Failed to activate parent widget of %s', ki.filename)
+		--	return false
+		--end
+	end
```

Comment:
Should this enable-parent-section go in the `if not ki.active then` block below? I'm not actually sure which; I guess it shouldn't be possible to have the child enabled but not the parent, so it might not matter?

Reply:
it is logically inconsistent/weird to have it outside that code block
I'll move it